src/screens/Inventory/AddInventory.tsx

```diff
@@ -0,0 +1,93 @@
+import { createStyles, FormControl, InputLabel, MenuItem, Select, Theme } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+import BaseScreen from '../../components/BaseComponents/BaseScreen';
+import Button from '../../components/Button';
+import TextField from '../../components/TextField';
+import { SiteRecord } from '../../lib/airtable/interface';
+import { createInventory } from '../../lib/airtable/request';
+import { EMPTY_INVENTORY } from '../../lib/redux/inventoryDataSlice';
+import { EMPTY_SITE } from '../../lib/redux/siteDataSlice';
+import { RootState } from '../../lib/redux/store';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    content: {
+      color: theme.palette.text.primary,
+    },
+    formControl: {
+      width: '100%',
+    },
+  });
+
+interface AddInventoryProps extends RouteComponentProps {
+  classes: { content: string, formControl: string};
+  location: any;
+  currentSite: SiteRecord;
+  products: any[];
+  userId: string;
+}
+
+
+function AddInventory(props: AddInventoryProps) {
+  const { classes, currentSite, products, userId} = props;
+  const history = useHistory();
+
+  const [selectedProductId, setSelectedProductId] = useState("");
+  const [startingAmount, setStartingAmount] = useState(0);
+
+  // TODO: Add form input validation and error messaging
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    // Make a deep copy of an empty inventory record
+    const inventory = JSON.parse(JSON.stringify(EMPTY_INVENTORY));
+    inventory.productId = selectedProductId;
+    inventory.siteId = currentSite.id;
+    inventory.currentQuantity = startingAmount;
+    inventory.periodStartQuantity = startingAmount;
+
+    createInventory(inventory);
```

Comment:
Like the AddCustomer functionality, we also want to save separately to redux in case of offline. This can be another PR if you want, but it'd be good to put a TODO in that case